Tasks/AzureCLIV1/azureclitask.ts

```diff
@@ -116,37 +117,40 @@ export class azureclitask {
     private static servicePrincipalId: string = null;
     private static servicePrincipalKey: string = null;
 
-    private static loginAzure() {
-        var connectedService: string = tl.getInput("connectedServiceNameARM", true);
-        this.loginAzureRM(connectedService);
-    }
-
     private static loginAzureRM(connectedService: string): void {
-        var servicePrincipalId: string = tl.getEndpointAuthorizationParameter(connectedService, "serviceprincipalid", false);
-        let authType: string = tl.getEndpointAuthorizationParameter(connectedService, 'authenticationType', true);
-        let cliPassword: string = null;
-        if (authType == "spnCertificate") {
-            tl.debug('certificate based endpoint');
-            let certificateContent: string = tl.getEndpointAuthorizationParameter(connectedService, "servicePrincipalCertificate", false);
-            cliPassword = path.join(tl.getVariable('Agent.TempDirectory') || tl.getVariable('system.DefaultWorkingDirectory'), 'spnCert.pem');
-            fs.writeFileSync(cliPassword, certificateContent);
-            this.cliPasswordPath = cliPassword;
+        var authScheme: string = tl.getEndpointAuthorizationScheme(connectedService, true);
+        var subscriptionID: string = tl.getEndpointDataParameter(connectedService, "SubscriptionID", true);
 
+        if(authScheme == "ServicePrincipal") {
```

Comment:
Do we need to do case insensitive comparison?

Reply:
Done